pynumaflow/reducer/servicer/async_servicer.py

```diff
@@ -66,109 +62,72 @@ async def ReduceFn(
-        """
 
-        start, end = None, None
-        for metadata_key, metadata_value in context.invocation_metadata():
-            if metadata_key == WIN_START_TIME:
-                start = metadata_value
-            elif metadata_key == WIN_END_TIME:
-                end = metadata_value
-        if not (start or end):
-            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
-            context.set_details(
-                f"Expected to have all key/window_start_time/window_end_time; "
-                f"got start: {start}, end: {end}."
-            )
-            yield reduce_pb2.ReduceResponse(results=[])
-            return
-
-        start_dt = datetime.fromtimestamp(int(start) / 1e3, timezone.utc)
-        end_dt = datetime.fromtimestamp(int(end) / 1e3, timezone.utc)
-        interval_window = IntervalWindow(start=start_dt, end=end_dt)
+        The lifecycle of the reduce operation is managed by the task manager. Which is created
+        for each new grpc call.
+
+        Whenever a new request is received, the task manager invokes the required
+        operation based on the request type.
+        1) If the request is for a new keyed window, a new task is created.
+           If the request is for an existing keyed window, the data is
+           appended to the task.
 
+        2) Once the request iterator is exhausted, the task manager sends EOF to all
+        the tasks, to signal them to stop reading the data from their respective iterators.
+
+        3) The task manager then waits for the tasks to complete and
+        yields the response from each task
+
+        """
+        # Create an async iterator from the request iterator
         datum_iterator = datum_generator(request_iterator=request_iterator)
 
-        response_task = asyncio.create_task(
-            self.__async_reduce_handler(interval_window, datum_iterator)
-        )
+        # Create a task manager instance
+        # The task manager is used to manage lifecycle of the tasks
+        # required for the reduce operation.
+        task_manager = TaskManager(handler=self.__reduce_handler)
 
-        # Save a reference to the result of this function, to avoid a
-        # task disappearing mid-execution.
-        self.background_tasks.add(response_task)
-        response_task.add_done_callback(lambda t: self.background_tasks.remove(t))
+        # Start iterating through the request iterator and create tasks
+        # based on the operation type received.
+        try:
+            async for request in datum_iterator:
+                # check whether the request is an open or append operation
+                if request.operation is int(WindowOperation.OPEN):
+                    # create a new task for the open operation
+                    await task_manager.create_task(request)
+                elif request.operation is int(WindowOperation.APPEND):
+                    # append the task data to the existing task
```

Comment:
```suggestion
                    # append the task data to the existing task
                    # incase task is missing, it will create a task too.
```